src/pheromones/hivemind-host/src/HiveMindHostSerializer.cpp

```diff
@@ -0,0 +1,22 @@
+#include "HiveMindHostSerializer.h"
+#include <pb_encode.h>
+
+HiveMindHostSerializer::HiveMindHostSerializer(IProtobufStream& stream) : m_stream(stream) {}
+
+bool HiveMindHostSerializer::serialize(const MessageDTO& message) {
+
+    Message msg_send;
+    message.serialize(msg_send);
+
+    pb_ostream_t outputStream{HiveMindHostSerializer::streamCallback, this, SIZE_MAX, 0, 0};
+
+    return pb_encode_ex(&outputStream, Message_fields, &msg_send, PB_ENCODE_DELIMITED);
```

Comment:
pb_encode is blocking, it needs all the bytes to encode/decode. The IProtobuf stream needs to provide enough bytes for protobuf. So ex if the callback asks for 100 bytes, the IProtobufStream needs to probides them (by blocking) or return false. So the callback are not asynchronous. 

Now if IProtobuf stream whants to create an intermediary buffer to queue bytes before pb asks for them (so it's faster) that could be possible. So it would "cache" request before it gets asked for them